benchexec/container.py

```diff
@@ -507,29 +509,45 @@ def duplicate_mount_hierarchy(mount_base, temp_base, work_base, dir_modes):
             work_path = work_base + b"/" + str(overlay_count).encode()
             os.makedirs(temp_path, exist_ok=True)
             os.makedirs(work_path, exist_ok=True)
-            try:
-                # Previous mount in this place not needed if replaced with overlay dir.
-                libc.umount(mount_path)
-            except OSError as e:
-                logging.debug(e)
+            if os.path.ismount(mount_path):
+                try:
+                    # Previous mount in this place not needed if replaced with overlay dir.
+                    libc.umount(mount_path)
+                except OSError as e:
+                    logging.debug(e)
             try:
                 make_overlay_mount(mount_path, mountpoint, temp_path, work_path)
             except OSError as e:
-                mp = mountpoint.decode()
-                raise OSError(
-                    e.errno,
-                    f"Creating overlay mount for '{mp}' failed: {os.strerror(e.errno)}. "
-                    f"Please use other directory modes, "
-                    f"for example '--read-only-dir {util.escape_string_shell(mp)}'.",
-                )
+                # Resort to fuse-overlayfs if kernel overlayfs is not available.
+                fuse = util.find_executable2("fuse-overlayfs")
+                if fuse:
+                    logging.debug(
+                        "Cannot use kernel overlay for %s: %s. "
+                        "Trying to use fuse-overlayfs instead.",
+                        mountpoint.decode(),
+                        e,
+                    )
+                    cap_permitted_to_ambient()
```

Comment:
Do we need this here for every single overlay mount?

Reply:
No, this is only needed when using `fuse-overlayfs`. Moving it to a more general location would make it seem universal, but in fact, only `fuse-overlayfs` requires it.